src/nvim/event/rstream.c

```diff
@@ -187,6 +188,7 @@ static void read_event(void **argv)
   if (stream->read_cb) {
     size_t count = (uintptr_t)argv[1];
     bool eof = (uintptr_t)argv[2];
+    stream->did_eof = eof;
     stream->read_cb(stream, stream->buffer, count, stream->cb_data, eof);
```

Comment:
it is not the same information. `eof` is part of the data relayed to the callback, it is not an "extra argument".  `stream->did_eof` is to avoid sending a superfluous EOF callback in the job cleanup code (which ensures all output callbacks are sent before the exit callback).